File: v-lang/l-java/v-spring/v-web-interceptor/src/main/java/org/example/controller/PingController.java
package org.example.controller;

import org.example.anno.Foo;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class PingController {

    @GetMapping("/ping")
    public String ping() {
        return "pong";
    }

    @Foo
    @GetMapping("/foo")
    public String foo() {
        return "foo";
    }

    @GetMapping("/bar")
    public String bar() {
        return "bar";
    }
}

File: v-demo/demo-sso/sso-spring/src/main/kotlin/org/example/controller/PingController.kt
package org.example.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class PingController {

    @RequestMapping("/ping")
    fun ping(): ResponseEntity<String> = ResponseEntity.ok("pong...")
}